### PR TITLE
Add full-text search to order filtering

### PR DESCRIPTION
**Pull request by patrys**
Force simple config for all search operations to opt-out of stemming and stop words.

Implements SALEOR-6508

QA: need to verify pagination with new UUID-based order IDs as we make a lot of assumptions inside filtering and pagination code.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6508 Full-text search for orders](https://app.clickup.com/t/2549495/SALEOR-6508)

**Review by IKarbowiak — APPROVED** [review]

File: saleor/checkout/complete_checkout.py [hunk 1]
```diff
@@ -1068,7 +1070,8 @@ def _create_order_from_checkout(
     update_order_authorize_data(order, with_save=False)
 
     # order search
-    order.search_document = prepare_order_search_document_value(order)
+    order.search_vector = prepare_order_search_vector_value(order)
+    print(">>>", order.search_vector)
```
Comment (fowczarek) [inline saleor/checkout/complete_checkout.py:1074]: IMO it's not necessary.

File: saleor/graphql/order/tests/test_order.py [hunk 2]
```diff
@@ -8960,8 +8947,9 @@ def test_draft_orders_query_with_filter_search(
         ]
     )
     for order in orders:
-        order.search_document = prepare_order_search_document_value(order)
-    Order.objects.bulk_update(orders, ["search_document"])
+        order.search_vector = prepare_order_search_vector_value(order)
+        print(">>>", order.search_vector)
```
Comment (fowczarek) [inline saleor/graphql/order/tests/test_order.py:8951]: IMO it's not necessary.

**Review by fowczarek — CHANGES_REQUESTED** [review]

File: saleor/order/models.py [hunk 3]
```diff
@@ -287,7 +288,7 @@ class Order(ModelWithMetadata):
     )
     redirect_url = models.URLField(blank=True, null=True)
     search_document = models.TextField(blank=True, default="")
-
+    search_vector = SearchVectorField(blank=True, null=True)
```
Comment (fowczarek) [inline saleor/order/models.py:291]: Should we remove `search_document` because we remove the function responsible for updating it?

Comment (patrys) [reply saleor/order/models.py:291]: I don't want to break the database during upgrade, it was also done like that for products.

**Review by fowczarek — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**